src/components/OrgFile/components/AgendaModal/index.js

```diff
@@ -172,12 +176,14 @@ const mapStateToProps = (state) => {
   const file = state.org.present.getIn(['files', path]);
   const allFiles = state.org.present.get('files');
   const fileSettings = state.org.present.get('fileSettings');
+  const agendaStartOnWeekday = state.base.get('agendaStartOnWeekday');
   return {
     files: determineIncludedFiles(allFiles, fileSettings, path, 'includeInAgenda', false),
     todoKeywordSets: file.get('todoKeywordSets'),
     agendaTimeframe: state.base.get('agendaTimeframe'),
     agendaDefaultDeadlineDelayValue: state.base.get('agendaDefaultDeadlineDelayValue') || 5,
     agendaDefaultDeadlineDelayUnit: state.base.get('agendaDefaultDeadlineDelayUnit') || 'd',
+    agendaStartOnWeekday: agendaStartOnWeekday == null ? 1 : +agendaStartOnWeekday,
```

Comment:
Why not `===`? I thought it's best practice to use it instead of `==`. Why the `+`? To convert to integer?

How far do we trust the state store in terms of data integrity? The existing lines above this one don't check the types. So I think we better stick to that scheme. @munen what do you think?

Reply:
Right, triple equals is best practice in general but some consider double equals against null and undefined to be an exception since it’s a convenient way of eliminating both at once. Yes + is for number conversion (maybe this was too hacky)

I’m type checking here because falseness check interferes with 0... I’ll comment more on that in #677